### ISSUE TITLE
krings may not be properly aligned

### ISSUE THREAD

**Issue by jcaplan** [luigirizzo/netmap#921]
https://github.com/luigirizzo/netmap/blob/b58a473becf02c03db4504bbd66987ef5b2fc3d8/sys/dev/netmap/netmap.c#L877

Depending on the values of `[n]` (specifically if `n[0]` + `n[1]` is odd), then `kring` may not be properly aligned.

**Comment by giuseppelettieri** [comment]
You are right. tailroom is only used in VALE with a large enough alignment, but better to be explicit. I have pushed a new new commit for this.

**Comment by jcaplan** [comment]
I actually encountered this when tailroom passed into the function is 0. We maintain a port of FreeBSD network stack to QNX. when we upgraded to GCC12 we started getting segfaults running ctrl-api-test due to an optimization using FP registers and simd instruction on unaligned kring address. 

For us the solution was to check that `kring % __alignof__(*kring) == 0`. 

Your solution would not have fixed the problem for us.

**Comment by giuseppelettieri** [comment]
mmm, OK, that's an entirely different story. Those instructions want 16-byte alignment (old wisdom was that FP registers were off-limits in the kernel, but things change). I'll see what I can do.

**Comment by jcaplan** [comment]
QNX is a micro kernel so we’re running in user space
